pkg/provider/gcp/secretmanager/provider.go

```diff
@@ -65,10 +66,11 @@ func (p *Provider) NewClient(ctx context.Context, store esv1beta1.GenericStore,
 	useMu.Lock()
 
 	client := &Client{
-		kube:      kube,
-		store:     gcpStore,
-		storeKind: store.GetKind(),
-		namespace: namespace,
+		kube:        kube,
+		store:       gcpStore,
+		storeKind:   store.GetKind(),
+		namespace:   namespace,
+		secretLocks: locks.SecretLocks{},
```

Comment:
We potentially can have many `&Client{}` instances at the same time, when ESO reconciles multiple resources concurrently. (not configurable at the moment)

In order to lock things across `&Client{}` instances we need a shared `SecretLock`. Further, we'll need a `provider` prefix in the SecretLock, as it may be used across providers (AWS/GCP...).